### ISSUE TITLE
Store non-object files/dirs into a separeted disk from ones for object files

### ISSUE THREAD

**Issue by tmenjo** [sheepdog/sheepdog#205]
Current Sheepdog stores not only object files but also other files
and dirs e.g. config and epoch into the disk passed to sheep command,
even if using multi-disk feature.

Doing like as below:

    # sheep ... /path/to/disk0
       :
    # dog cluster format ...
    # dog node md plug ... /path/to/disk1
       :
    # dog create vdi ...
       :

then the disk0 will be "special" one:

    /path/to/disk0
    /path/to/disk0/config
    /path/to/disk0/epoch
    /path/to/disk0/epoch/00000001
    /path/to/disk0/lock
    /path/to/disk0/sheep.log
    /path/to/disk0/sock
    /path/to/disk0/obj
    /path/to/disk0/obj/.stale
    /path/to/disk0/obj/0016595b00000000
    /path/to/disk0/obj/003f1cb100000000
     :
    /path/to/disk1
    /path/to/disk1/.stale
    /path/to/disk1/0016595b00000001
    /path/to/disk1/004875e000000001
     :

The disks plugged with dog command e.g. disk1 are not "special".
If some of them crashes or are unplugged, the object files
stored there will be recovered from replicas in other nodes.

On the other hand, we will lose special files/dirs when the disk0 crashes,
then the sheep will be stopped. So we should still design its reliability
carefully, for instance, using RAID. I don't think it is user-friendly.

I think it is good that object files and non-object files are stored in
the separated disks. Then we can assign a reliable disk only for the former,
and it may be same as the one for OS (which generally forms RAID1).

Now, a new option would be required:

    # sheep ... --new-option=/path/to/reliable-disk /path/to/disk0
       :
    # dog cluster format ...
    # dog node md plug /path/to/disk1
       :
    # dog create vdi ...
       :

to store files/dirs separately like as below:

    /path/to/reliable-disk
    /path/to/reliable-disk/config
    /path/to/reliable-disk/epoch
    /path/to/reliable-disk/epoch/00000001
    /path/to/reliable-disk/lock
    /path/to/reliable-disk/sheep.log
    /path/to/reliable-disk/sock
     :
    /path/to/disk0/
    /path/to/disk0/.stale
    /path/to/disk0/0016595b00000000
    /path/to/disk0/003f1cb100000000
     :
    /path/to/disk1
    /path/to/disk1/.stale
    /path/to/disk1/0016595b00000001
    /path/to/disk1/004875e000000001

**Comment by sirio81** [comment]
2016-01-07 11:03 GMT+01:00 Takashi Menjo <notifications@github.com>:

> Current Sheepdog stores not only object files but also other files
> and dirs e.g. config and epoch into the disk passed to sheep command,
> even if using multi-disk feature.
>
> Doing like as below:
>
> # sheep ... /path/to/disk0
>
>
A new option is not needed.
You can simply start sheep like this:

# sheep /var/lib/sheepdog, /path/to/disk0

/var/lib/sheepdog is on the / of you os, that it's probably on a raid.
If /var/lib/sheepdog is not accessible it mean your os has trouble and so
the whole node.
Off course you can choose any other path than /var/lib/sheepdog.
Mind that the first directory you mention is for metadata.
If you specify only one directory, this will contain metadata and the
objects in the 'obj' subdirectory as you notices.
In other words, always specify at least two directories.

**Comment by tmenjo** [comment]
@sirio81:

Thank you for your comment.
I first know that I can specify two or more dirs with sheep command...

I started a sheep process with multi directories. The snippet is below.
Metadata files and object files can be stored in separated disks!

    # mkdir -p /var/lib/sheepdog/{log,disk0,disk1,disk2}
    
    # /usr/sbin/sheep -l dir=/var/lib/sheepdog/log,level=debug /var/lib/sheepdog/disk0,/var/lib/sheepdog/disk1,/var/lib/sheepdog/disk2
    
    # ps -ef | grep -E '^UID|/usr/sbin/sheep' | grep -v grep
    UID        PID  PPID  C STIME TTY          TIME CMD
    root     12761  2061  0 13:38 ?        00:00:00 /usr/sbin/sheep -l dir=/var/lib/sheepdog/log level=debug /var/lib/sheepdog/disk0 /var/lib/sheepdog/disk1,/var/lib/sheepdog/disk2
    root     12762 12761  0 13:38 ?        00:00:00 /usr/sbin/sheep -l dir=/var/lib/sheepdog/log level=debug /var/lib/sheepdog/disk0 /var/lib/sheepdog/disk1,/var/lib/sheepdog/disk2
    
    # dog cluster format -c 1
    
    # dog node md info
    
    Id  Size    Used    Avail   Use%    Path
    0   2.7 TB  0.0 MB  2.7 TB    0%    /var/lib/sheepdog/disk1
    1   2.7 TB  0.0 MB  2.7 TB    0%    /var/lib/sheepdog/disk2
    
    # find /var/lib/sheepdog
    /var/lib/sheepdog
    /var/lib/sheepdog/log
    /var/lib/sheepdog/log/sheep.log
    /var/lib/sheepdog/disk0
    /var/lib/sheepdog/disk0/lock
    /var/lib/sheepdog/disk0/obj
    /var/lib/sheepdog/disk0/epoch
    /var/lib/sheepdog/disk0/epoch/00000001
    /var/lib/sheepdog/disk0/config
    /var/lib/sheepdog/disk0/sock
    /var/lib/sheepdog/disk1
    /var/lib/sheepdog/disk1/.stale
    /var/lib/sheepdog/disk2
    /var/lib/sheepdog/disk2/.stale
    
    (create some VDIs here)
    
    # find /var/lib/sheepdog
    /var/lib/sheepdog
    /var/lib/sheepdog/log
    /var/lib/sheepdog/log/sheep.log
    /var/lib/sheepdog/disk0
    /var/lib/sheepdog/disk0/lock
    /var/lib/sheepdog/disk0/obj
    /var/lib/sheepdog/disk0/epoch
    /var/lib/sheepdog/disk0/epoch/00000001
    /var/lib/sheepdog/disk0/config
    /var/lib/sheepdog/disk0/sock
    /var/lib/sheepdog/disk1
    /var/lib/sheepdog/disk1/.stale
    /var/lib/sheepdog/disk1/00700bc000000000
    /var/lib/sheepdog/disk1/000e86e400000001
    /var/lib/sheepdog/disk1/8063747300000000
    /var/lib/sheepdog/disk1/8092b7e900000000
    /var/lib/sheepdog/disk1/001e070800000001
    /var/lib/sheepdog/disk1/805f29f100000000
    /var/lib/sheepdog/disk1/80b75ba500000000
    /var/lib/sheepdog/disk1/00b75ba500000001
    /var/lib/sheepdog/disk1/0091613300000000
    /var/lib/sheepdog/disk1/0091613300000001
    /var/lib/sheepdog/disk1/0065d79c00000000
    /var/lib/sheepdog/disk2
    /var/lib/sheepdog/disk2/.stale
    /var/lib/sheepdog/disk2/80700bc000000000
    /var/lib/sheepdog/disk2/00700bc000000001
    /var/lib/sheepdog/disk2/800e86e400000000
    /var/lib/sheepdog/disk2/000e86e400000000
    /var/lib/sheepdog/disk2/0063747300000000
    /var/lib/sheepdog/disk2/0063747300000001
    /var/lib/sheepdog/disk2/0092b7e900000000
    /var/lib/sheepdog/disk2/0092b7e900000001
    /var/lib/sheepdog/disk2/801e070800000000
    /var/lib/sheepdog/disk2/001e070800000000
    /var/lib/sheepdog/disk2/005f29f100000000
    /var/lib/sheepdog/disk2/005f29f100000001
    /var/lib/sheepdog/disk2/00b75ba500000000
    /var/lib/sheepdog/disk2/8091613300000000
    /var/lib/sheepdog/disk2/8065d79c00000000
    /var/lib/sheepdog/disk2/0065d79c00000001

I'll try some more things to understand sheep's behavior deeply.

**Comment by tmenjo** [comment]
I started a sheep process with one disk i.e. disk0 then plugged another disk i.e. disk1.
After that, I unplugged **disk0/obj** correctly then found that the sheep process did not become down
and metadata files/dirs e.g. config and epoch were still stored in disk0.

So I understand that the best practice is 1. or 2.:

 1. Start sheep process with multiple disks, reliable one is for metadata, the other(s) for object
 2. Start sheep process with one reliable disk, afterwards plug/unplug disk(s) properly

I will close this issue later. Thanks for telling me, @sirio81 !

### snippet (a bit long):

    # mkdir -p /var/lib/sheepdog/1/{log,disk0,disk1}
    # mkdir -p /var/lib/sheepdog/2/{log,disk0,disk1}
    # /usr/sbin/sheep -l dir=/var/lib/sheepdog/1/log,level=debug -p 7001 -z 1 /var/lib/sheepdog/1/disk0
    # /usr/sbin/sheep -l dir=/var/lib/sheepdog/2/log,level=debug -p 7002 -z 2 /var/lib/sheepdog/2/disk0
    
    # ps -ef | grep -E '^UID|/usr/sbin/sheep' | grep -v grep
    UID        PID  PPID  C STIME TTY          TIME CMD
    root     13448  2061  0 14:50 ?        00:00:00 /usr/sbin/sheep -l dir=/var/lib/sheepdog/1/log level=debug -p 7001 -z 1 /var/lib/sheepdog/1/disk0
    root     13449 13448  0 14:50 ?        00:00:00 /usr/sbin/sheep -l dir=/var/lib/sheepdog/1/log level=debug -p 7001 -z 1 /var/lib/sheepdog/1/disk0
    root     13465  2061  0 14:50 ?        00:00:00 /usr/sbin/sheep -l dir=/var/lib/sheepdog/2/log level=debug -p 7002 -z 2 /var/lib/sheepdog/2/disk0
    root     13466 13465  0 14:50 ?        00:00:00 /usr/sbin/sheep -l dir=/var/lib/sheepdog/2/log level=debug -p 7002 -z 2 /var/lib/sheepdog/2/disk0
    
    # dog node list -p 7001
      Id   Host:Port         V-Nodes       Zone
       0   127.0.0.1:7001       128          1
       1   127.0.0.1:7002       128          2
    
    # dog cluster format -p 7001 -c 2
    
    # find /var/lib/sheepdog
    /var/lib/sheepdog
    /var/lib/sheepdog/1
    /var/lib/sheepdog/1/log
    /var/lib/sheepdog/1/log/sheep.log
    /var/lib/sheepdog/1/disk0
    /var/lib/sheepdog/1/disk0/lock
    /var/lib/sheepdog/1/disk0/obj
    /var/lib/sheepdog/1/disk0/obj/.stale
    /var/lib/sheepdog/1/disk0/epoch
    /var/lib/sheepdog/1/disk0/epoch/00000001
    /var/lib/sheepdog/1/disk0/config
    /var/lib/sheepdog/1/disk0/sock
    /var/lib/sheepdog/1/disk1
    /var/lib/sheepdog/2
    /var/lib/sheepdog/2/log
    /var/lib/sheepdog/2/log/sheep.log
    /var/lib/sheepdog/2/disk0
    /var/lib/sheepdog/2/disk0/lock
    /var/lib/sheepdog/2/disk0/obj
    /var/lib/sheepdog/2/disk0/obj/.stale
    /var/lib/sheepdog/2/disk0/epoch
    /var/lib/sheepdog/2/disk0/epoch/00000001
    /var/lib/sheepdog/2/disk0/config
    /var/lib/sheepdog/2/disk0/sock
    /var/lib/sheepdog/2/disk1
    
    (create some VDIs here)
    
    # find /var/lib/sheepdog
    /var/lib/sheepdog
    /var/lib/sheepdog/1
    /var/lib/sheepdog/1/log
    /var/lib/sheepdog/1/log/sheep.log
    /var/lib/sheepdog/1/disk0
    /var/lib/sheepdog/1/disk0/lock
    /var/lib/sheepdog/1/disk0/obj
    /var/lib/sheepdog/1/disk0/obj/.stale
    /var/lib/sheepdog/1/disk0/obj/80b91bda00000000
    /var/lib/sheepdog/1/disk0/obj/00b91bda00000000
    /var/lib/sheepdog/1/disk0/obj/00b91bda00000001
    /var/lib/sheepdog/1/disk0/obj/80be3af000000000
    /var/lib/sheepdog/1/disk0/obj/00be3af000000000
    /var/lib/sheepdog/1/disk0/obj/00be3af000000001
    /var/lib/sheepdog/1/disk0/obj/8051cf1900000000
    /var/lib/sheepdog/1/disk0/obj/0051cf1900000000
    /var/lib/sheepdog/1/disk0/obj/0051cf1900000001
    /var/lib/sheepdog/1/disk0/obj/80f9f47700000000
    /var/lib/sheepdog/1/disk0/obj/00f9f47700000000
    /var/lib/sheepdog/1/disk0/obj/00f9f47700000001
    /var/lib/sheepdog/1/disk0/obj/80cb01bf00000000
    /var/lib/sheepdog/1/disk0/obj/00cb01bf00000000
    /var/lib/sheepdog/1/disk0/obj/00cb01bf00000001
    /var/lib/sheepdog/1/disk0/epoch
    /var/lib/sheepdog/1/disk0/epoch/00000001
    /var/lib/sheepdog/1/disk0/config
    /var/lib/sheepdog/1/disk0/sock
    /var/lib/sheepdog/1/disk1
    /var/lib/sheepdog/2
    /var/lib/sheepdog/2/log
    /var/lib/sheepdog/2/log/sheep.log
    /var/lib/sheepdog/2/disk0
    /var/lib/sheepdog/2/disk0/lock
    /var/lib/sheepdog/2/disk0/obj
    /var/lib/sheepdog/2/disk0/obj/.stale
    /var/lib/sheepdog/2/disk0/obj/80b91bda00000000
    /var/lib/sheepdog/2/disk0/obj/00b91bda00000000
    /var/lib/sheepdog/2/disk0/obj/00b91bda00000001
    /var/lib/sheepdog/2/disk0/obj/80be3af000000000
    /var/lib/sheepdog/2/disk0/obj/00be3af000000000
    /var/lib/sheepdog/2/disk0/obj/00be3af000000001
    /var/lib/sheepdog/2/disk0/obj/8051cf1900000000
    /var/lib/sheepdog/2/disk0/obj/0051cf1900000000
    /var/lib/sheepdog/2/disk0/obj/0051cf1900000001
    /var/lib/sheepdog/2/disk0/obj/80f9f47700000000
    /var/lib/sheepdog/2/disk0/obj/00f9f47700000000
    /var/lib/sheepdog/2/disk0/obj/00f9f47700000001
    /var/lib/sheepdog/2/disk0/obj/80cb01bf00000000
    /var/lib/sheepdog/2/disk0/obj/00cb01bf00000000
    /var/lib/sheepdog/2/disk0/obj/00cb01bf00000001
    /var/lib/sheepdog/2/disk0/epoch
    /var/lib/sheepdog/2/disk0/epoch/00000001
    /var/lib/sheepdog/2/disk0/config
    /var/lib/sheepdog/2/disk0/sock
    /var/lib/sheepdog/2/disk1
    
    # dog node md info -p 7001
    Id  Size    Used    Avail   Use%    Path
     0  2.7 TB  40 MB   2.7 TB    0%    /var/lib/sheepdog/1/disk0/obj
    
    # dog node md info -p 7002
    Id  Size    Used    Avail   Use%    Path
     0  2.7 TB  40 MB   2.7 TB    0%    /var/lib/sheepdog/2/disk0/obj
    
    # dog node md plug -p 7001 /var/lib/sheepdog/1/disk1
        __
       ()'`;
       /\|`
      /  |   Caution! The directories to be plugged will be purged.
    (/_)_|_  Are you sure you want to continue? [yes/no]: yes
    
    # dog node md info -p 7001
    Id  Size    Used    Avail   Use%    Path
     0  2.7 TB  12 MB   2.7 TB    0%    /var/lib/sheepdog/1/disk0/obj
     1  2.7 TB  40 MB   2.7 TB    0%    /var/lib/sheepdog/1/disk1
    
    # find /var/lib/sheepdog
    /var/lib/sheepdog
    /var/lib/sheepdog/1
    /var/lib/sheepdog/1/log
    /var/lib/sheepdog/1/log/sheep.log
    /var/lib/sheepdog/1/disk0
    /var/lib/sheepdog/1/disk0/lock
    /var/lib/sheepdog/1/disk0/obj
    /var/lib/sheepdog/1/disk0/obj/.stale
    /var/lib/sheepdog/1/disk0/obj/80b91bda00000000
    /var/lib/sheepdog/1/disk0/obj/80be3af000000000
    /var/lib/sheepdog/1/disk0/obj/00be3af000000000
    /var/lib/sheepdog/1/disk0/obj/8051cf1900000000
    /var/lib/sheepdog/1/disk0/obj/0051cf1900000001
    /var/lib/sheepdog/1/disk0/obj/80f9f47700000000
    /var/lib/sheepdog/1/disk0/obj/00cb01bf00000001
    /var/lib/sheepdog/1/disk0/epoch
    /var/lib/sheepdog/1/disk0/epoch/00000001
    /var/lib/sheepdog/1/disk0/config
    /var/lib/sheepdog/1/disk0/sock
    /var/lib/sheepdog/1/disk1
    /var/lib/sheepdog/1/disk1/.stale
    /var/lib/sheepdog/1/disk1/0051cf1900000000
    /var/lib/sheepdog/1/disk1/00be3af000000001
    /var/lib/sheepdog/1/disk1/00b91bda00000000
    /var/lib/sheepdog/1/disk1/00b91bda00000001
    /var/lib/sheepdog/1/disk1/00cb01bf00000000
    /var/lib/sheepdog/1/disk1/80cb01bf00000000
    /var/lib/sheepdog/1/disk1/00f9f47700000001
    /var/lib/sheepdog/1/disk1/00f9f47700000000
    /var/lib/sheepdog/2
    /var/lib/sheepdog/2/log
    /var/lib/sheepdog/2/log/sheep.log
    /var/lib/sheepdog/2/disk0
    /var/lib/sheepdog/2/disk0/lock
    /var/lib/sheepdog/2/disk0/obj
    /var/lib/sheepdog/2/disk0/obj/.stale
    /var/lib/sheepdog/2/disk0/obj/80b91bda00000000
    /var/lib/sheepdog/2/disk0/obj/00b91bda00000000
    /var/lib/sheepdog/2/disk0/obj/00b91bda00000001
    /var/lib/sheepdog/2/disk0/obj/80be3af000000000
    /var/lib/sheepdog/2/disk0/obj/00be3af000000000
    /var/lib/sheepdog/2/disk0/obj/00be3af000000001
    /var/lib/sheepdog/2/disk0/obj/8051cf1900000000
    /var/lib/sheepdog/2/disk0/obj/0051cf1900000000
    /var/lib/sheepdog/2/disk0/obj/0051cf1900000001
    /var/lib/sheepdog/2/disk0/obj/80f9f47700000000
    /var/lib/sheepdog/2/disk0/obj/00f9f47700000000
    /var/lib/sheepdog/2/disk0/obj/00f9f47700000001
    /var/lib/sheepdog/2/disk0/obj/80cb01bf00000000
    /var/lib/sheepdog/2/disk0/obj/00cb01bf00000000
    /var/lib/sheepdog/2/disk0/obj/00cb01bf00000001
    /var/lib/sheepdog/2/disk0/epoch
    /var/lib/sheepdog/2/disk0/epoch/00000001
    /var/lib/sheepdog/2/disk0/config
    /var/lib/sheepdog/2/disk0/sock
    /var/lib/sheepdog/2/disk1
    
    # dog node md unplug -p 7001 /var/lib/sheepdog/1/disk0/obj
        __
       ()'`;
       /\|`
      /  |   Caution! Recovery will be triggered after unplugging.
    (/_)_|_  Are you sure you want to continue? [yes/no]: yes
    
    # dog node md info -p 7001
    Id  Size    Used    Avail   Use%    Path
     0  2.7 TB  52 MB   2.7 TB    0%    /var/lib/sheepdog/1/disk1
    
    # find /var/lib/sheepdog
    /var/lib/sheepdog
    /var/lib/sheepdog/1
    /var/lib/sheepdog/1/log
    /var/lib/sheepdog/1/log/sheep.log
    /var/lib/sheepdog/1/disk0
    /var/lib/sheepdog/1/disk0/lock
    /var/lib/sheepdog/1/disk0/obj
    /var/lib/sheepdog/1/disk0/obj/.stale
    /var/lib/sheepdog/1/disk0/obj/80b91bda00000000
    /var/lib/sheepdog/1/disk0/obj/80be3af000000000
    /var/lib/sheepdog/1/disk0/obj/00be3af000000000
    /var/lib/sheepdog/1/disk0/obj/8051cf1900000000
    /var/lib/sheepdog/1/disk0/obj/0051cf1900000001
    /var/lib/sheepdog/1/disk0/obj/80f9f47700000000
    /var/lib/sheepdog/1/disk0/obj/00cb01bf00000001
    /var/lib/sheepdog/1/disk0/epoch
    /var/lib/sheepdog/1/disk0/epoch/00000001
    /var/lib/sheepdog/1/disk0/config
    /var/lib/sheepdog/1/disk0/sock
    /var/lib/sheepdog/1/disk1
    /var/lib/sheepdog/1/disk1/.stale
    /var/lib/sheepdog/1/disk1/0051cf1900000000
    /var/lib/sheepdog/1/disk1/00be3af000000001
    /var/lib/sheepdog/1/disk1/00b91bda00000000
    /var/lib/sheepdog/1/disk1/00b91bda00000001
    /var/lib/sheepdog/1/disk1/00cb01bf00000000
    /var/lib/sheepdog/1/disk1/80cb01bf00000000
    /var/lib/sheepdog/1/disk1/00f9f47700000001
    /var/lib/sheepdog/1/disk1/00f9f47700000000
    /var/lib/sheepdog/1/disk1/0051cf1900000001
    /var/lib/sheepdog/1/disk1/00be3af000000000
    /var/lib/sheepdog/1/disk1/00cb01bf00000001
    /var/lib/sheepdog/1/disk1/8051cf1900000000
    /var/lib/sheepdog/1/disk1/80b91bda00000000
    /var/lib/sheepdog/1/disk1/80be3af000000000
    /var/lib/sheepdog/1/disk1/80f9f47700000000
    /var/lib/sheepdog/2
    /var/lib/sheepdog/2/log
    /var/lib/sheepdog/2/log/sheep.log
    /var/lib/sheepdog/2/disk0
    /var/lib/sheepdog/2/disk0/lock
    /var/lib/sheepdog/2/disk0/obj
    /var/lib/sheepdog/2/disk0/obj/.stale
    /var/lib/sheepdog/2/disk0/obj/80b91bda00000000
    /var/lib/sheepdog/2/disk0/obj/00b91bda00000000
    /var/lib/sheepdog/2/disk0/obj/00b91bda00000001
    /var/lib/sheepdog/2/disk0/obj/80be3af000000000
    /var/lib/sheepdog/2/disk0/obj/00be3af000000000
    /var/lib/sheepdog/2/disk0/obj/00be3af000000001
    /var/lib/sheepdog/2/disk0/obj/8051cf1900000000
    /var/lib/sheepdog/2/disk0/obj/0051cf1900000000
    /var/lib/sheepdog/2/disk0/obj/0051cf1900000001
    /var/lib/sheepdog/2/disk0/obj/80f9f47700000000
    /var/lib/sheepdog/2/disk0/obj/00f9f47700000000
    /var/lib/sheepdog/2/disk0/obj/00f9f47700000001
    /var/lib/sheepdog/2/disk0/obj/80cb01bf00000000
    /var/lib/sheepdog/2/disk0/obj/00cb01bf00000000
    /var/lib/sheepdog/2/disk0/obj/00cb01bf00000001
    /var/lib/sheepdog/2/disk0/epoch
    /var/lib/sheepdog/2/disk0/epoch/00000001
    /var/lib/sheepdog/2/disk0/config
    /var/lib/sheepdog/2/disk0/sock
    /var/lib/sheepdog/2/disk1

**Comment by sirio81** [comment]
2016-01-08 8:53 GMT+01:00 Takashi Menjo <notifications@github.com>:

> I started a sheep process with one disk i.e. disk0 then plugged another
> disk i.e. disk1.
> After that, I unplugged *disk0/obj* correctly then found that the sheep
> process did not become down
> and metadata files/dirs e.g. config and epoch were still stored in disk0.
>
> So I understand that the best practice is 1. or 2.:
>
>    1. Start sheep process with multiple disks, reliable one is for
>    metadata, the other(s) for object
>    2. Start sheep process with one reliable disk, afterwards plug/unplug
>    disk(s) properly
>
>
Metadata disk can never be unplugged.
In theory that action should kill the node, but probably is not handled by
sheepdog.
About point 2., if your intention is *not* to store object in the first
reliable device, it's not a good idea to start sheep with a single
directory because objects will be written in there.

I usually run sheep like this:

sheep /var/lib/sheepdog,/mnt/sheep/0,/mnt/sheep/1,/mnt/sheep/n

/var/lib/sheepdog is not a mount point, so it's on the /
/mnt/sheep/0 is the mount point for the first disk
/mnt/sheep/1 is the mount point for the second disk
/mnt/sheep/n and so on

**Comment by tmenjo** [comment]
> About point 2., if your intention is *not* to store object in the first
> reliable device, it's not a good idea to start sheep with a single
> directory because objects will be written in there.

I agree that it is not a good idea to use sheep with one directory.

In some cases, however, the users might have started sheep
in that way without intent. So I think " data migration" feature
from metadata disk to object disk would be needed for them.
This is what I want to try in my previous comment.

So I should not have said the way in point 2 as "best" practice...

> I usually run sheep like this:
> (snipped)

Thank you for telling your best practice! Your way fits what I think the best.
I'll try multi-disk feature using multiple disks actually in the near future.

**Comment by sirio81** [comment]
2016-01-12 2:48 GMT+01:00 Takashi Menjo <notifications@github.com>:

> In some cases, however, the users might have started sheep
> in that way without intent. So I think " data migration" feature
> from metadata disk to object disk would be needed for them.
>
If you have the possibility to stop the cluster, you can do that manually.



Considering you started sheep in the wrong path

sheep /wrong/path

and /var/lib/sheepdog and /mnt/sheep/0 to be the right path, this is the
procedure:

dog cluster shutdown
mv /wrong/path/obj/* /mnt/sheep/0
mv /wrong/path/obj/.stale /mnt/sheep/0
mv /wrong/path /var/lib/sheepdog

Then you have to start sheep with the right parameter

sheep /var/lib/sheepdog,/mnt/sheep/0
